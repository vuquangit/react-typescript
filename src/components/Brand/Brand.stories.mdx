import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks'
import { action } from '@storybook/addon-actions'

import AppWrapper from '@/test/supports/AppWrapper'
import Brand from '.'
import Space from 'stories/components/styled-system/Space.mdx'
import Styles from 'stories/styles.mdx'

<Meta title="Story/Components/Brand" component={Brand} />

<Styles />

# Brand

Logo site

export const Template = (args) => (
  <AppWrapper>
    <Brand {...args} />
  </AppWrapper>
)

<Canvas>
  <Story
    name="Normal"
    args={{ variant: 'normal' }}
    onClick={action('Medium Brand clicked')}
  >
    {Template.bind({})}
  </Story>
</Canvas>
