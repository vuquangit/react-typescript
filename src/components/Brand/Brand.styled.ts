import { Link } from 'react-router-dom'
import styled from 'styled-components'
import { space } from 'styled-system'

export const BrandWrapper = styled('div')`
  ${space};

  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;

  margin-left: -32px;
  margin-right: 24px;
`

export const BrandLink = styled(Link)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`

export const BrandLogo = styled('img')`
  width: 24px;
  height: 24px;
`
export const BrandName = styled('div')`
  margin-left: 8px;
`
