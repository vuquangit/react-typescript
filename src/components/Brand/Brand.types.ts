import { SpaceProps } from 'styled-system'

export declare type BaseBrandProps = SpaceProps
