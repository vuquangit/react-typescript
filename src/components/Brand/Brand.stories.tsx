import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import AppWrapper from '@/test/supports/AppWrapper'
import Brand from '.'

export default {
  title: 'Story/Components/Brand',
  component: Brand,
  argTypes: {},
} as Meta

const Template: Story = (args) => (
  <AppWrapper>
    <Brand {...args}>Brand story</Brand>
  </AppWrapper>
)

export const Normal = Template.bind({})
