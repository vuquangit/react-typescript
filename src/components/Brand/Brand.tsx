import React, { FC } from 'react'

import LogoImage from '@/assets/images/logo.jpg'
import { BrandWrapper, BrandLogo, BrandName, BrandLink } from './Brand.styled'
import { BaseBrandProps } from './Brand.types'

const Brand: FC<BaseBrandProps> = () => {
  return (
    <BrandWrapper>
      <BrandLink to="/">
        <BrandLogo src={LogoImage} alt="" />
        <BrandName>React</BrandName>
      </BrandLink>
    </BrandWrapper>
  )
}

export default Brand
