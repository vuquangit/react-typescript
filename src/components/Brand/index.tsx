/**
 *
 * Index for Brand
 *
 */

import Brand from './Brand'

export default Brand
