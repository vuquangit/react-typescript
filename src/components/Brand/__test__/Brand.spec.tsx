/**
 *
 * Tests for Brand
 *
 */

import React from 'react'

import AppWrapper from '@/test/supports/AppWrapper'
import Brand from '..'
import { shallow } from 'enzyme'

describe('App Brand', () => {
  it('Test default Brand', () => {
    const wrapper = shallow(
      <AppWrapper>
        <Brand />
      </AppWrapper>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
