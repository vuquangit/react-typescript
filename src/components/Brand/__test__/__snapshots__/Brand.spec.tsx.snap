// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Brand Test default Brand 1`] = `
<BrowserRouter>
  <MockedProvider
    addTypename={false}
    cache={
      InMemoryCache {
        "addTypename": true,
        "config": Object {
          "addTypename": true,
          "dataIdFromObject": [Function],
          "resultCaching": true,
          "typePolicies": Object {
            "Query": Object {
              "fields": Object {
                "isLoggedIn": Object {
                  "read": [Function],
                },
                "themeMode": Object {
                  "read": [Function],
                },
                "visibilityFilter": Object {
                  "read": [Function],
                },
              },
            },
          },
        },
        "data": Root {
          "canRead": [Function],
          "data": Object {},
          "getFieldValue": [Function],
          "group": CacheGroup {
            "caching": true,
            "d": [Function],
            "keyMaker": KeyTrie {
              "makeData": [Function],
              "weakness": true,
            },
          },
          "policies": Policies {
            "cache": [Circular],
            "config": Object {
              "cache": [Circular],
              "dataIdFromObject": [Function],
              "possibleTypes": undefined,
              "typePolicies": Object {
                "Query": Object {
                  "fields": Object {
                    "isLoggedIn": Object {
                      "read": [Function],
                    },
                    "themeMode": Object {
                      "read": [Function],
                    },
                    "visibilityFilter": Object {
                      "read": [Function],
                    },
                  },
                },
              },
            },
            "fuzzySubtypes": Map {},
            "rootIdsByTypename": Object {
              "Mutation": "ROOT_MUTATION",
              "Query": "ROOT_QUERY",
              "Subscription": "ROOT_SUBSCRIPTION",
            },
            "rootTypenamesById": Object {
              "ROOT_MUTATION": "Mutation",
              "ROOT_QUERY": "Query",
              "ROOT_SUBSCRIPTION": "Subscription",
            },
            "supertypeMap": Map {},
            "typePolicies": Object {
              "Query": Object {
                "fields": Object {
                  "isLoggedIn": Object {
                    "keyFn": undefined,
                    "merge": undefined,
                    "read": [Function],
                  },
                  "themeMode": Object {
                    "keyFn": undefined,
                    "merge": undefined,
                    "read": [Function],
                  },
                  "visibilityFilter": Object {
                    "keyFn": undefined,
                    "merge": undefined,
                    "read": [Function],
                  },
                },
                "keyFn": undefined,
              },
            },
            "usingPossibleTypes": false,
          },
          "refs": Object {},
          "rootIds": Object {},
          "sharedLayerGroup": CacheGroup {
            "caching": true,
            "d": [Function],
            "keyMaker": KeyTrie {
              "makeData": [Function],
              "weakness": true,
            },
          },
          "storageTrie": KeyTrie {
            "makeData": [Function],
            "weakness": true,
          },
          "toReference": [Function],
        },
        "getFragmentDoc": [Function],
        "makeVar": [Function],
        "maybeBroadcastWatch": [Function],
        "optimisticData": Root {
          "canRead": [Function],
          "data": Object {},
          "getFieldValue": [Function],
          "group": CacheGroup {
            "caching": true,
            "d": [Function],
            "keyMaker": KeyTrie {
              "makeData": [Function],
              "weakness": true,
            },
          },
          "policies": Policies {
            "cache": [Circular],
            "config": Object {
              "cache": [Circular],
              "dataIdFromObject": [Function],
              "possibleTypes": undefined,
              "typePolicies": Object {
                "Query": Object {
                  "fields": Object {
                    "isLoggedIn": Object {
                      "read": [Function],
                    },
                    "themeMode": Object {
                      "read": [Function],
                    },
                    "visibilityFilter": Object {
                      "read": [Function],
                    },
                  },
                },
              },
            },
            "fuzzySubtypes": Map {},
            "rootIdsByTypename": Object {
              "Mutation": "ROOT_MUTATION",
              "Query": "ROOT_QUERY",
              "Subscription": "ROOT_SUBSCRIPTION",
            },
            "rootTypenamesById": Object {
              "ROOT_MUTATION": "Mutation",
              "ROOT_QUERY": "Query",
              "ROOT_SUBSCRIPTION": "Subscription",
            },
            "supertypeMap": Map {},
            "typePolicies": Object {
              "Query": Object {
                "fields": Object {
                  "isLoggedIn": Object {
                    "keyFn": undefined,
                    "merge": undefined,
                    "read": [Function],
                  },
                  "themeMode": Object {
                    "keyFn": undefined,
                    "merge": undefined,
                    "read": [Function],
                  },
                  "visibilityFilter": Object {
                    "keyFn": undefined,
                    "merge": undefined,
                    "read": [Function],
                  },
                },
                "keyFn": undefined,
              },
            },
            "usingPossibleTypes": false,
          },
          "refs": Object {},
          "rootIds": Object {},
          "sharedLayerGroup": CacheGroup {
            "caching": true,
            "d": [Function],
            "keyMaker": KeyTrie {
              "makeData": [Function],
              "weakness": true,
            },
          },
          "storageTrie": KeyTrie {
            "makeData": [Function],
            "weakness": true,
          },
          "toReference": [Function],
        },
        "policies": Policies {
          "cache": [Circular],
          "config": Object {
            "cache": [Circular],
            "dataIdFromObject": [Function],
            "possibleTypes": undefined,
            "typePolicies": Object {
              "Query": Object {
                "fields": Object {
                  "isLoggedIn": Object {
                    "read": [Function],
                  },
                  "themeMode": Object {
                    "read": [Function],
                  },
                  "visibilityFilter": Object {
                    "read": [Function],
                  },
                },
              },
            },
          },
          "fuzzySubtypes": Map {},
          "rootIdsByTypename": Object {
            "Mutation": "ROOT_MUTATION",
            "Query": "ROOT_QUERY",
            "Subscription": "ROOT_SUBSCRIPTION",
          },
          "rootTypenamesById": Object {
            "ROOT_MUTATION": "Mutation",
            "ROOT_QUERY": "Query",
            "ROOT_SUBSCRIPTION": "Subscription",
          },
          "supertypeMap": Map {},
          "typePolicies": Object {
            "Query": Object {
              "fields": Object {
                "isLoggedIn": Object {
                  "keyFn": undefined,
                  "merge": undefined,
                  "read": [Function],
                },
                "themeMode": Object {
                  "keyFn": undefined,
                  "merge": undefined,
                  "read": [Function],
                },
                "visibilityFilter": Object {
                  "keyFn": undefined,
                  "merge": undefined,
                  "read": [Function],
                },
              },
              "keyFn": undefined,
            },
          },
          "usingPossibleTypes": false,
        },
        "storeReader": StoreReader {
          "config": Object {
            "addTypename": true,
            "cache": [Circular],
          },
          "executeSelectionSet": [Function],
          "executeSubSelectedArray": [Function],
          "knownResults": WeakMap {},
        },
        "storeWriter": StoreWriter {
          "cache": [Circular],
          "reader": StoreReader {
            "config": Object {
              "addTypename": true,
              "cache": [Circular],
            },
            "executeSelectionSet": [Function],
            "executeSubSelectedArray": [Function],
            "knownResults": WeakMap {},
          },
        },
        "txCount": 0,
        "typenameDocumentCache": Map {},
        "watchDep": [Function],
        "watches": Set {},
      }
    }
    mocks={
      Array [
        Object {
          "request": Object {
            "query": Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "getTheme",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [
                          Object {
                            "arguments": Array [],
                            "kind": "Directive",
                            "name": Object {
                              "kind": "Name",
                              "value": "client",
                            },
                          },
                        ],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "themeMode",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 46,
                "start": 0,
              },
            },
          },
          "result": Object {
            "data": Object {
              "themeMode": "light",
            },
          },
        },
      ]
    }
  >
    <Component
      theme={Object {}}
    >
      <Memo(l) />
      <Brand />
    </Component>
  </MockedProvider>
</BrowserRouter>
`;
